test/ng/filter/filtersSpec.js

```diff
@@ -86,6 +86,20 @@ describe('filters', function() {
       expect(num).toBe('123.100');
     });
 
+    it('should work with negative fractionSize', function() {
+      expect(formatNumber(1234, pattern, ',', '.', -1)).toBe("1,230");
+      expect(formatNumber(1234.567, pattern, ',', '.', -1)).toBe("1,230");
+      expect(formatNumber(1235, pattern, ',', '.', -1)).toBe("1,240");
+      expect(formatNumber(1235, pattern, ',', '.', -2)).toBe("1,200");
+      expect(formatNumber(1235, pattern, ',', '.', -3)).toBe("1,000");
+      expect(formatNumber(1235, pattern, ',', '.', -4)).toBe("0");
+      expect(formatNumber(1236, pattern, ',', '.', -1)).toBe("1,240");
+      expect(formatNumber(1000, pattern, ',', '.', -1)).toBe("1,000");
+      expect(formatNumber(1, pattern, ',', '.', -1)).toBe("0");
+      expect(formatNumber(1, pattern, ',', '.', -2)).toBe("0");
+      expect(formatNumber(9, pattern, ',', '.', -1)).toBe("10");
```

Comment:
Although the behavior of negative `fractionSize` isn't defined somewhere (afaik), there are some cases that behave differently that I expect:

**(A)**
```js
expect(formatNumber(1000, pattern, ',', '.', -4)).toBe("0");   // But the actual value is `0.000`
```

and

**(B)**
```js
expect(formatNumber(51, pattern, ',', '.', -2)).toBe('100');      // But the actual value is `10`
expect(formatNumber(501, pattern, ',', '.', -3)).toBe('1,000');   // But the actual value is `10`
```

Basically, there are two issues (both appearing when using a negative `fractionSize` whose absolute value equals the number of integer digit in the number):

* **(A)**: When the number ends with zeros (as in `1000`), unneeded "decimal" zeros are appended at the end of the result.

* **(B)**: When the first digit of the number is >= 5, it is always rounded up to `10`.